docs/about/architecture.md

```diff
@@ -268,3 +268,53 @@ Coder on Kubernetes.
   [Microsoft Entra ID Sign-On](https://learn.microsoft.com/en-us/entra/identity/app-proxy/)
 - For GCP:
   [Google Cloud Identity Platform](https://cloud.google.com/architecture/identity/single-sign-on)
+
+### Devcontainers
+
+Note: _Devcontainers_ are at early stage and considered experimental at the
+moment.
+
+This architecture enhances a Coder workspace with
+[Devcontainers](https://containers.dev/) setup built using
+[envbuilder](https://github.com/coder/envbuilder) project. Workspace users have
+the flexibility to extend generic, base developer environments with custom,
+project-oriented [features](https://containers.dev/features) without requiring
+platform administrators to push altered Docker images.
+
+Learn more about
+[Devcontainers support](https://coder.com/docs/v2/latest/templates/devcontainers)
+in Coder.
+
+![Architecture Diagram](../images/architecture-devcontainers.png)
+
+#### Components
+
+The deployment model includes:
+
+- _Workspace_ built using Coder template with _envbuilder_ enabled to set up the
+  developer environment accordingly to the spec.
+- _Container Registry_ for Docker images used by _envbuilder_, maintained by
+  Coder platform engineers or developer productivity engineers.
+
+Since this model is strictly focused on workspace nodes, it does not affect the
+setup of regional infrastructure. It can be deployed alongside other deployment
+models, in multiple regions, or across various cloud platforms.
+
+##### Workload resources
+
+**Workspace**
+
+- Docker and Kubernetes based templates are supported.
+- The `docker_container` resource uses `ghcr.io/coder/envbuilder` as base image.
+
+The _envbuilder_ checks out the base Docker image from the container registry,
+and applies extra feature layers, and starts the container with target
```

Comment:
```suggestion
and applies extra feature layers, and starts the container with the target
```

Nit: I'm not sure what this is trying to convey. What's the "target environment"? As a reader I might also wonder what these extra feature layers are.

Perhaps target environment means environment as specified in `devcontainer.json`? If so, I think we could improve the wording.

Note: The dev container specification doesn't mention layers in any significant proportion. There's a tiny note down on this page, for instance: https://containers.dev/implementors/features/. Hence the concept may not be clear. Perhaps "layer" is more of an implementation detail, not something users need to care about/know? (Fine to keep in the graph/image though.)

Reply:
Good call, rephrased! I used the word layers as this is mostly what envbuilder does. It builds next layers = features on top of the base image.